### DISCUSSION TITLE
tRPC?

### DISCUSSION THREAD

**Discussion by cope** [withastro/roadmap#264] [Proposal]
Since [TypeScript](https://docs.astro.build/en/guides/typescript/) is supported, and [fetch](https://docs.astro.build/en/guides/data-fetching/) handles [GraphQL queries](https://docs.astro.build/en/guides/data-fetching/#graphql-queries), it would be cool to also add 
[tRPC](https://trpc.io/) support, maybe, hopefully? 🙂

**Comment by reggi** [comment]
You could theoretically use whatever you want. I'm not sure how you'd use tRPC with Astro.

**Comment by matthewp** [comment]
You should be able to use TRPC today. Is there some sort of integration you'd like to see that makes it easier?

**Reply by reggi** [reply]
Put more eloquently then me 👆

**Reply by mayank99** [reply]
"you should be able to use tRPC today" is correct. https://github.com/trpc/trpc/discussions/2123#discussioncomment-3452622

**Reply by matthewp** [reply]
That's really cool. It should be possible to create a tRPC integration that uses [inject route](https://docs.astro.build/en/reference/integrations-reference/#injectroute-option) so that apps don't need to define the routes themselves.

**Reply by cope** [reply]
> You should be able to use TRPC today. Is there some sort of integration you'd like to see that makes it easier?

THB, I haven't yet figured out how to use @astrojs/node even 😞
